### PR TITLE
✨ New component: [amp-access-fewcents] paywall support by Fewcents

### PR DESCRIPTION
**Pull request by mayank-s-dev**
**Ref to I2I**: https://github.com/ampproject/amphtml/issues/31854

**Description:**
The amp-access-fewcents component is based amp-access, it is built for amp pages which internally uses amp-access to provide a behavior similar to amp-access, but built in such a way that it can be used with Fewcents.

This component also relies on Access Content Markup to show and hide content. It also gives option to publisher's to have their logo on the paywall and pass their config.

In the current implementation we have given you the skeleton of the component by the authorize API call and rendering only few parts of component.

**References:**

We have been following [amp-access-laterpay](https://github.com/ampproject/amphtml/tree/main/extensions/amp-access-laterpay) and [amp-access-poool](https://github.com/ampproject/amphtml/tree/main/extensions/amp-access-poool) for our reference.
- **[amp-access](https://github.com/ampproject/amphtml/tree/main/extensions/amp-access)**
- **AMP code contribution docs:** https://github.com/ampproject/amphtml/blob/main/docs/contributing-code.md
- **First time setup:** https://github.com/ampproject/amphtml/blob/main/docs/getting-started-e2e.md
- **Building an extension:** https://github.com/ampproject/amphtml/blob/main/docs/building-an-amp-extension.md

### CODE REVIEW

File: examples/amp-access-fewcents.html
Comment (twifkak) [inline examples/amp-access-fewcents.html:1]: Should this example use the new amp-access-fewcents component?

File: extensions/amp-access-fewcents/amp-access-fewcents.md
Comment (twifkak) [inline extensions/amp-access-fewcents/amp-access-fewcents.md:1]: It looks like this doc is not fully finished. Some TODO markers in it, and references to Bento which I'm not sure apply in this case? I'm not sure what is the best practice here. @calebcordry should we wait to add a document until ready, or is there a way to mark it as "draft"?

**Review by twifkak — APPROVED** [review]
Thanks for the contribution! Just reviewing for @ampproject/wg-caching, the protoascii and validator html test file look good to me; approving for those files only. I asked a couple of questions on some auxiliary files on which I'm not the expert and would appreciate input from some others. Please wait on others to review the rest of the files.

**Comment by mayank-s-dev** [comment]
Thank you for your approval @twifkak.

We are planning to complete our extension by this Christmas, we have completed the extension at our level, and raising PR's step by step with few lines of code. This has been advised by @calebcordry.

File: examples/amp-access-fewcents.html
Comment (calebcordry) [reply examples/amp-access-fewcents.html:1]: these access components are a bit weird in that they don't actually have their own `<amp-access-xxx>` custom elements. You might have been confused by validator file, which I think is incorrect.

File: examples/amp-access-fewcents.html
Comment (calebcordry) [inline examples/amp-access-fewcents.html:20]: This isn't doing anything?

File: extensions/amp-access-fewcents/0.1/amp-access-fewcents.js
Comment (calebcordry) [inline extensions/amp-access-fewcents/0.1/amp-access-fewcents.js:27]: nit: I think this can be `access-vendor.AccessVendor`

File: extensions/amp-access-fewcents/0.1/fewcents-impl.js
Comment (calebcordry) [inline extensions/amp-access-fewcents/0.1/fewcents-impl.js:138]: nit: we have a few helpers for creating elements in `src/core/dom/index.js` you can use when appropriate.

File: extensions/amp-access-fewcents/amp-access-fewcents.md
Comment (calebcordry) [reply extensions/amp-access-fewcents/amp-access-fewcents.md:1]: +1 Lets add docs at the very end when everything is fleshed out? Or at least mark it as draft as suggested.

File: extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii
Comment (calebcordry) [inline extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii:11]: I don't think we actually need this?

File: extensions/amp-access-fewcents/0.1/storybook/Basic.amp.js
Comment (calebcordry) [inline extensions/amp-access-fewcents/0.1/storybook/Basic.amp.js:1]: I think we can delete this file as it for bento which this is not?

File: extensions/amp-access-fewcents/0.1/test/validator-amp-access-fewcents.html
Comment (calebcordry) [inline extensions/amp-access-fewcents/0.1/test/validator-amp-access-fewcents.html:19]: Is this the real dom structure? If not lets remove.

**Review by calebcordry — CHANGES_REQUESTED** [review]

**Comment by calebcordry** [comment]
Thanks for breaking this up! Just a few comments.

File: examples/amp-access-fewcents.html
Comment (twifkak) [reply examples/amp-access-fewcents.html:1]: Ah, OK, thanks.

File: examples/amp-access-fewcents.html
Comment (mayank-s-dev) [reply examples/amp-access-fewcents.html:20]: This has been auto generated while creating the extension template, I will check on this and update it.

File: extensions/amp-access-fewcents/0.1/fewcents-impl.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/fewcents-impl.js:138]: I will have a look at them, and update it.

File: extensions/amp-access-fewcents/0.1/storybook/Basic.amp.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/storybook/Basic.amp.js:1]: It is not required to us as from my understanding, so I am removing it.

File: extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii:11]: > Thanks for breaking this up! Just a few comments.

Thanks @calebcordry for your review, I will make the changes you asked and let you know.

File: extensions/amp-access-fewcents/amp-access-fewcents.md
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/amp-access-fewcents.md:1]: I will mark it as draft.

File: extensions/amp-access-fewcents/0.1/amp-access-fewcents.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/amp-access-fewcents.js:27]: Will update it.

File: extensions/amp-access-fewcents/0.1/storybook/Basic.amp.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/storybook/Basic.amp.js:1]: I exactly didn't understand bento, but I don't think it is required to us. So removing it.

**Comment by mayank-s-dev** [comment]
> Thanks for breaking this up! Just a few comments.

Thanks @calebcordry for your review, I will make the changes you asked and let you know.

File: extensions/amp-access-fewcents/0.1/test/validator-amp-access-fewcents.html
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/test/validator-amp-access-fewcents.html:19]: Will remove this and add the real dom structure.

File: extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii:11]: Could you please help me here, I haven't understood why this file is required?

File: examples/amp-access-fewcents.html
Comment (mayank-s-dev) [reply examples/amp-access-fewcents.html]: Changes done.

File: extensions/amp-access-fewcents/0.1/amp-access-fewcents.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/amp-access-fewcents.js]: Changes done.

File: extensions/amp-access-fewcents/0.1/fewcents-impl.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/fewcents-impl.js]: Changes done.

File: extensions/amp-access-fewcents/0.1/storybook/Basic.amp.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/storybook/Basic.amp.js]: Changes done.

File: extensions/amp-access-fewcents/0.1/test/validator-amp-access-fewcents.html
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/test/validator-amp-access-fewcents.html]: Changes done.

**Comment by mayank-s-dev** [comment]
> Thanks for breaking this up! Just a few comments.

@calebcordry I have done the changes, except one, for which I have asked your help in above comment.

File: extensions/amp-access-fewcents/0.1/fewcents-impl.js
Comment (calebcordry) [inline extensions/amp-access-fewcents/0.1/fewcents-impl.js:63]: nit: unnecessary comment for self documenting function name.

File: extensions/amp-access-fewcents/0.1/fewcents-impl.js
Comment (calebcordry) [inline extensions/amp-access-fewcents/0.1/fewcents-impl.js:165]: this code might be cleaner if you just use `createElementWithAttributes(ampdoc, 'div', {className: foo})`

Comment (calebcordry) [reply extensions/amp-access-fewcents/0.1/fewcents-impl.js:165]: or pass attributes in your helper method.

File: extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii
Comment (calebcordry) [reply extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii:11]: its required for the validator to recognize a new extension e.g. `<amp-access-fewcents>` but since this is just a extension of the `<amp-access>` component I think we can just delete this file.

File: extensions/amp-access-fewcents/amp-access-fewcents.md
Comment (calebcordry) [inline extensions/amp-access-fewcents/amp-access-fewcents.md:15]: remove this comment :)

File: extensions/amp-access-fewcents/amp-access-fewcents.md
Comment (calebcordry) [reply extensions/amp-access-fewcents/amp-access-fewcents.md:1]: @twifkak do you know how to actually mark this md as draft? I am not sure

File: extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/validator-amp-access-fewcents.protoascii:11]: Okay I will delete it.

File: extensions/amp-access-fewcents/amp-access-fewcents.md
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/amp-access-fewcents.md]: Changes done.

File: extensions/amp-access-fewcents/0.1/fewcents-impl.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/fewcents-impl.js]: Changes done.

File: extensions/amp-access-fewcents/0.1/fewcents-impl.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/fewcents-impl.js]: Changes done.

**Comment by mayank-s-dev** [comment]
@calebcordry I have done the recent changes you asked.

File: extensions/amp-access-fewcents/OWNERS
Comment (calebcordry) [inline extensions/amp-access-fewcents/OWNERS:7]: can you change this to wg-monetization? You should also add yourself.

File: extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/0.1/test/test-amp-access-fewcents.js]: Updated.

File: extensions/amp-access-fewcents/OWNERS
Comment (mayank-s-dev) [reply extensions/amp-access-fewcents/OWNERS]: Done.

**Comment by mayank-s-dev** [comment]
Thank you @calebcordry for your approval.

**Review by newmuis — APPROVED** [review]
Approved, for OWNERS

**Comment by mayank-s-dev** [comment]
Hey @calebcordry @newmuis @twifkak

I see the merging is blocked, could you guys help me out how I can resolve these errors to unblock it?

**Comment by mayank-s-dev** [comment]
> Approved, for OWNERS

Thank you @newmuis

**Comment by calebcordry** [comment]
@mayank-s-dev looks like there are still a few CI errors. Can you try to resolve them ?https://app.circleci.com/pipelines/github/ampproject/amphtml/20730/workflows/e374b9cf-f264-493e-965a-8253f476d87b

Specifically the validation errors, the e2e tests are probably just flaky. Let me know if you have any issues, thanks.

**Comment by mayank-s-dev** [comment]
> @mayank-s-dev looks like there are still a few CI errors. Can you try to resolve them ?https://app.circleci.com/pipelines/github/ampproject/amphtml/20730/workflows/e374b9cf-f264-493e-965a-8253f476d87b
> 
> Specifically the validation errors, the e2e tests are probably just flaky. Let me know if you have any issues, thanks.

@calebcordry I have fix the CI errors, to fix them I need to remove a line whose source is currently not available. I am planning to add it in next cut of PR. You can find that line here.
`<script async custom-element="amp-access-fewcents" src="https://cdn.ampproject.org/v0/amp-access-fewcents-0.1.js"></script>`

Also I need your help to resolve the bundle-size issue [here](https://github.com/ampproject/amphtml/pull/37096/checks?check_run_id=4593107030)

**Comment by calebcordry** [comment]
Thanks! Looks like we are having issues with the bundle size bot unrelated to this PR. I will let you know if any action is needed.

**Comment by mayank-s-dev** [comment]
> Thanks! Looks like we are having issues with the bundle size bot unrelated to this PR. I will let you know if any action is needed.

Sure, Thank you.

**Comment by mayank-s-dev** [comment]
Hey Guys, @calebcordry @newmuis @twifkak,

Any update on this?

**Comment by mayank-s-dev** [comment]
@calebcordry we are really eager to move forward on to next set of steps, do let us know if any action is required from our end.

**Comment by mayank-s-dev** [comment]
@calebcordry @newmuis @twifkak 
Hey Guys, any update on this PR? we have started preparing the second cut of this PR.

**Comment by calebcordry** [comment]
Sorry folks I was out of office for a bit. Nice work.

**Comment by siddharthlshah** [comment]
@calebcordry No worries! Happy New Year and thanks a lot for merging the PR!!

**Comment by mayank-s-dev** [comment]
Thank you @calebcordry.

**Comment by mayank-s-dev** [comment]
@calebcordry @newmuis @twifkak 
Hey All, while checking for releases we found our extension is a part of this [release](https://github.com/ampproject/amphtml/releases/tag/2201071715001). 
When we can expect the release of this PR?

**Comment by calebcordry** [comment]
> @calebcordry @newmuis @twifkak Hey All, while checking for releases we found our extension is a part of this [release](https://github.com/ampproject/amphtml/releases/tag/2201071715001). When we can expect the release of this PR?

This PR should hit prod today or tomorrow unless release is delayed for some reason. You should be able to test in beta now. There is more detail about the AMP release here.

**Comment by mayank-s-dev** [comment]
> > @calebcordry @newmuis @twifkak Hey All, while checking for releases we found our extension is a part of this [release](https://github.com/ampproject/amphtml/releases/tag/2201071715001). When we can expect the release of this PR?
> 
> This PR should hit prod today or tomorrow unless release is delayed for some reason. You should be able to test in beta now. There is more detail about the AMP release here https://github.com/ampproject/amphtml/blob/9d7cbd80413bde9b978bcce4b27515dc4ca17302/docs/release-schedule.md

Thanks @calebcordry, PR has been released.

**Pull request merged**
